### PR TITLE
Add infrastructure nodes sizing recommendations

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit adds information on infrastructure nodes which hosts
Router, Registry, Logging and Monitoring components for various
nodes scale. The data is based on the cluster maximums and density
focuded scale test runs.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
@ahardin-rh @jtaleric @mffiedler PTAL.

File: modules/infrastructure-node-sizing.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/recommended-host-practices.adoc
+
+[id="infrastructure-node-sizing_{context}"]
+=  Infrastructure node sizing
+
+The infrastructure node resource requirements depend on the cluster age, number of nodes and objects in the cluster as they will lead to increase in the number of metrics or time series in Prometheus. The following infrastructure node size recommendations are based on the results of cluster maximums and control plane density focused testing.
```
Comment (ahardin-rh) [inline modules/infrastructure-node-sizing.adoc:8]: nodes, and objects in the cluster, as these factors can lead to an increase in the number of metrics

File: modules/infrastructure-node-sizing.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+| 500
+| 32
+| 192
+
+|===
+
+[IMPORTANT]
+====
+The sizing recommendations are based on the scale tests which create large number of objects across the cluster including hitting some of the cluster maximums 10000 Namespaces along with 61000 Pods, 10000 Deployments, 181000 Secrets, 400 ConfigMaps e.t.c. in case of 250 and 500 node counts on a {product-title} {product-version} cluster. Prometheus is a highly memory intensive application, the resource usage depends on various factors including the number of nodes, objects, prometheus metrics scraping interval, metrics or time series and age of the cluster while the disk size also depends on the retention period. You must take these factors in to consideration and size them accordingly.
```
Comment (ahardin-rh) [inline modules/infrastructure-node-sizing.adoc:34]: These sizing recommendations are based on scale tests, which create a large number of objects across the cluster. These tests include reaching some of the cluster maximums. In the case of  250 and 500 node counts on a {product-title} {product-version} cluster, these maximums are 10000 namespaces with 61000 Pods, 10000 deployments, 181000 secrets, 400 ConfigMaps, and so on. Prometheus is a highly memory intensive application; the resource usage depends on various factors including the number of nodes, objects, the Prometheus metrics scraping interval, metrics or time series, and the age of the cluster. The disk size also depends on the retention period. You must take these factors into consideration and size them accordingly.

File: modules/infrastructure-node-sizing.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+====
+
+[NOTE]
+====
+In {product-title} {product-version}, half of a CPU core (500 millicore) is now reserved by the system by default compared to {product-title} 3.11 and previous versions. The sizes are determined taking that into consideration.
+====
```
Comment (ahardin-rh) [inline modules/infrastructure-node-sizing.adoc:40]: s/The sizes are determined taking that into consideration./This influences the stated sizing recommendations.

**Comment by ahardin-rh** [comment]
@chaitanyaenr Is this for 4.5?

**Comment by chaitanyaenr** [comment]
@ahardin-rh yes, this is for 4.5.

**Comment by ahardin-rh** [comment]
@mffiedler Can you please provide QE review? Thanks!

**Comment by mffiedler** [comment]
@chaitanyaenr If we are recommending ES and Prometheus to share infra nodes, we should revisit the memory/core numbers one more time.   Should have raised this earlier...

**Comment by chaitanyaenr** [comment]
@mffiedler We can just consider the default infrastructure components which gets set up during the install ( Prometheus, Router and Registry ) for sizing and revisit the case when logging is installed on infra nodes like you mentioned. Added a note in the PR. PTAL.

**Comment by ahardin-rh** [comment]
@mffiedler @chaitanyaenr Is this good to go? Thanks!

File: modules/infrastructure-node-sizing.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+| 32
+| 192
+
+|===
+
+[IMPORTANT]
+====
+These sizing recommendations are based on scale tests, which create a large number of objects across the cluster. These tests include reaching some of the cluster maximums. In the case of 250 and 500 node counts on a {product-title} {product-version} cluster, these maximums are 10000 namespaces with 61000 Pods, 10000 deployments, 181000 secrets, 400 ConfigMaps, and so on. Prometheus is a highly memory intensive application; the resource usage depends on various factors including the number of nodes, objects, the Prometheus metrics scraping interval, metrics or time series, and the age of the cluster. The disk size also depends on the retention period. You must take these factors into consideration and size them accordingly.
+
+The sizing recommendations are applicable only for the infrastructure components which gets installed during the cluster install - Prometheus, Router and Registry. Logging is a day two operation and the recommendations does not take it in to account.
```
Comment (mffiedler) [inline modules/infrastructure-node-sizing.adoc:36]: s/the recommendations does not take it into account/the recommendations do not take it in to account/   or into, not sure which is proper.

Comment (chaitanyaenr) [reply modules/infrastructure-node-sizing.adoc]: Fixed it, thanks.

**Review by mffiedler — APPROVED** [review]
/LGTM

**Pull request merged**